From f33a738fd7bb8b5e1148b0a5424778db7342450b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Thu, 22 Oct 2026 10:07:20 +0200
Subject: [PATCH] BD-J: do not advance getMediaTime() at rate 0

getMediaTime() extrapolates the media time by the time base whenever the
player is Started and ignores the rate, while getMediaNanoseconds() stops
at rate 0. An application reading the time through
MediaTimePositionControl sees it run during a pause. Build the Time from
getMediaNanoseconds().
---
 .../bdj/java/org/videolan/media/content/BDHandler.java     | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/libbluray/bdj/java/org/videolan/media/content/BDHandler.java b/src/libbluray/bdj/java/org/videolan/media/content/BDHandler.java
index 2d5c700..dc56a51 100644
--- a/src/libbluray/bdj/java/org/videolan/media/content/BDHandler.java
+++ b/src/libbluray/bdj/java/org/videolan/media/content/BDHandler.java
@@ -194,12 +194,7 @@ public abstract class BDHandler implements Player, ServiceContentHandler {
     }
 
     public Time getMediaTime() {
-        synchronized (this) {
-            if (state == Started)
-                return new Time((baseMediaTime + (getTimeBase().getNanoseconds() - baseTime)) / 1000000000.0d);
-            else
-                return new Time(baseMediaTime / 1000000000.0d);
-        }
+        return new Time(getMediaNanoseconds() / 1000000000.0d);
     }
 
     public long getMediaNanoseconds() {
-- 
2.53.0

